packages/kotti-ui/source/kotti-field/types.ts

```diff
@@ -38,135 +83,166 @@ export namespace KottiField {
 	 * These values can be explicitly overriden in the field usage
 	 */
-	export type InhertiableProps = {
+	export const inheritablePropsSchema = z.object({
 		/**
 		 * Is the field disabled?
 		 * This will e.g. prevent changing the value
 		 */
-		isDisabled: boolean
+		isDisabled: z.boolean().default(false),
 
 		/**
 		 * Show a skeleton in place of the field
 		 */
-		isLoading: boolean
+		isLoading: z.boolean().default(false),
 
 		/**
 		 * Should the user have the option to clear the field
 		 */
-		hideClear: boolean
+		hideClear: z.never(),
 
 		/**
 		 * Prevents the validation (e.g. color, text) from being shown
 		 */
-		hideValidation: boolean
+		hideValidation: z.boolean().default(false),
 
 		/**
 		 * Defines the size of the field which influences child styles
 		 * to make fields e.g. appear more compact
 		 */
-		size: Size | null
-	}
-
+		size: z.nativeEnum(Size).default(Size.MEDIUM),
+	})
+	export type InhertiableProps = z.input<typeof inheritablePropsSchema>
 	/**
-	 * When adding a new prop, please make sure that no KtFormField
-	 * already uses a prop with the same name, to avoid conflicts
+	 * Warning: All values of type `never` are replaced with `any`
 	 */
-	export type Props<DATA_TYPE, PLACEHOLDER_TYPE> = InhertiableProps & {
+	export type InhertiablePropsInternal = ReplaceRecordType<
+		z.output<typeof inheritablePropsSchema>
+	>
+
+	export const propsSchema = inheritablePropsSchema.extend({
 		/**
 		 * Specifies that the data KtFormContext[formKey]
 		 * If formKey is "NONE", it is treated as an explicit opt-out
 		 * of the context-based behavior
 		 */
-		formKey: typeof FORM_KEY_NONE | string | null
+		formKey: z
+			.union([z.string(), z.literal(FORM_KEY_NONE)])
+			.nullable()
+			.default(null),
 
 		/**
 		 * Adds a small questionmark popover to the label to explain
 		 * more complicated forms to the user on-demand
 		 */
-		helpDescription: string | null
-
+		helpDescription: z.string().nullable().default(null),
 		/**
 		 * Adds a description below the label
 		 */
-		helpText: string | null
+		helpText: z.string().nullable().default(null),
 
 		/**
 		 * Shows a Yoco icon on the left side of the field
 		 */
-		leftIcon: Yoco.Icon | null
+		leftIcon: z.never(),
 
 		/**
 		 * Shown when no value was entered yet
 		 *
-		 * this is generic, as KtFieldDateRange needs two placeholders [string | null, string | null]
+		 * This is never, and specified by each component's schema, since they differ.
+		 * e.g. KtFieldDateRange needs two placeholders `[string | null, string | null]`
 		 */
-		placeholder: PLACEHOLDER_TYPE
+		placeholder: z.never(),
 
 		/**
 		 * Shows a Yoco icon on the right side of the field
 		 */
-		rightIcon: Yoco.Icon | null
+		rightIcon: z.never(),
 
-		isOptional: boolean
-		label: string | null
+		isOptional: z.boolean().default(false),
+		label: z.string().nullable().default(null),
 
 		/**
 		 * Show some string before the field
 		 */
-		prefix: string | null
+		prefix: z.never(),
```

Comment:
maybe leave a comment for 
`prefix`, `suffix`, `leftIcon`, `rightIcon`


to clarify that it's `never` and specified by the potentiallySupportedPropsSchema IF the component supports them